pkg/client/openstack/admin/client.go

```diff
@@ -251,3 +261,76 @@ func (c *adminClient) getRoleID(roleName string) (string, error) {
 	return "", fmt.Errorf("Role %s not found", roleName)
 
 }
+
+func (c *adminClient) CreateStorageContainer(projectID, containerName, serviceUserName, serviceUserDomainName string) error {
+	endpointURL, err := c.GetPublicObjectStoreEndpointURL(projectID)
+	if err != nil {
+		return err
+	}
+
+	storageClient, err := openstack.NewObjectStorageV1(c.ProviderClient, gophercloud.EndpointOpts{})
+	if err != nil {
+		return err
+	}
+	storageClient.Endpoint = endpointURL
+
+	domainID, err := c.getDomainID(serviceUserDomainName)
+	if err != nil {
+		return err
+	}
+
+	serviceUser, err := c.getUserByName(serviceUserName, domainID)
+	if err != nil {
+		return err
+	}
+
+	acl := fmt.Sprintf("%s:%s", projectID, serviceUser.ID)
+	createOpts := containers.CreateOpts{
+		ContainerRead:  acl,
+		ContainerWrite: acl,
+	}
+
+	_, err = containers.Create(storageClient, containerName, createOpts).Extract()
+
+	return err
+}
+
+func (c *adminClient) GetPublicObjectStoreEndpointURL(projectID string) (string, error) {
```

Comment:
Can you please make this a private function of the adminClient and remove it from the AdminClient interface for now?

Reply:
@databus23 I changed this.